src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2976,30 +2976,16 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
                     BaseCmdletPatterns = oldCmdletPatterns;
                 }
 
-                // If there is an existing session state and the new module info
-                // session state is empty, then use the existing session state.
-                // This will be the case when ModuleToProcess is a binary module
-                // and there were nested modules.
-                if (newManifestInfo.SessionState == null && ss != null)
```

Comment:
When it's not module analysis (get-module -list), `ss` will not be null. I verified that binary root module also use `ss` as the session state.